### PR TITLE
rose bunch: remove any existing database entries if first submit

### PR DESCRIPTION
**Pull request by arjclark**
Caters for a bug encountered in an odd situation.

If someone runs a bunch task in a suite and then later runs the suite and that same bunch task again (as a new first run), without having cleared out the work directory rose bunch will skip tasks it ran previously.

This fixes the bug. Test added here that should fail on master.

@benfitzpatrick - please review 1
@matthewrmshin - please review 2

### CODE REVIEW

File: lib/python/rose/apps/rose_bunch.py [hunk 1]
```diff
@@ -342,11 +342,16 @@ def connect(self):
     def create_tables(self):
         """Create tables as appropriate"""
         existing = []
+        first_run = os.environ["CYLC_TASK_SUBMIT_NUMBER"] == "1"
```
Comment (benfitzpatrick) [inline lib/python/rose/apps/rose_bunch.py]: Probably better if this is a '.get(`

**Comment by benfitzpatrick** [comment]
One comment, otherwise OK by me.

File: lib/python/rose/apps/rose_bunch.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/python/rose/apps/rose_bunch.py]: One problem with using the submit number is that it gets incremented for each *submit*, so if the first submit results in a submission failure, you will be in trouble here.

Comment (arjclark) [inline lib/python/rose/apps/rose_bunch.py]: @matthewrmshin - yeah I appreciate that but this is the best catch for it I can think of. Any better/other suggestions?

**Comment by arjclark** [comment]
@benfitzpatrick, change made. Over to you @matthewrmshin

**Pull request merged**